apis/duck/v1/status_types.go

```diff
@@ -76,9 +76,11 @@ func (*Conditions) GetFullType() ducktypes.Populatable {
 
 // GetCondition fetches the condition of the specified type.
 func (s *Status) GetCondition(t apis.ConditionType) *apis.Condition {
-	for _, cond := range s.Conditions {
+	for i, cond := range s.Conditions {
 		if cond.Type == t {
-			return &cond
```

Comment:
I think returning a copy is intentional to avoid having folks mutate conditions out from under us and violate the condition set semantics 🤔

Reply:
cc @n3wscott might know, or look at the commit history?

Reply:
This is fine if we intentionally return a copy, but we need to document that.